fiftyone/core/expression_ast.py

```
@@ -0,0 +1,644 @@
+    parts = [_render(n, fvar, evar)[0] for n in node.get("args", [])]
+    parts.extend(
+        "%s=%s" % (k, _render(v, fvar, evar)[0])
+        for k, v in node.get("kwargs", {}).items()
+    )
+    return ", ".join(parts)
+
+
+def _render_literal(node: LiteralNode) -> str:
+    as_ = node.get("as")
+    if as_ == "date":
+        return "datetime.utcfromtimestamp(%r / 1000)" % node["v"]
```

Ruff (0.15.21)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)